### PR TITLE
update syntax for site-region shortcodes

### PR DESCRIPTION
**Pull request by StefonSimmons**
### What does this PR do?

more updates to the  `site-region` shortcodes across docs.

updates `site-region` shortcode syntax on `en` pages:
- account-management
- integrations
- **agent**

hides toc items that reference `H2` headers on page load and **datadog site** region dropdown selection

### Motivation

https://datadoghq.atlassian.net/jira/software/projects/WEB/boards/102?selectedIssue=WEB-3279

### Preview 

syntax updates fix the TOC on the `agent` pages.

1.  `H3` headers show on TOC for `us1` region and don't show for other regions
- https://docs-staging.datadoghq.com/stefon.simmons/site-region-updates-2/agent/guide/private-link/?tab=useast1

2. `H2` and `H3` headers show on TOC for all regions except for `us1-fed`
- https://docs-staging.datadoghq.com/stefon.simmons/site-region-updates-2/agent/guide/how_remote_config_works/?tab=configurationyamlfile

### Additional Notes

related to: https://github.com/DataDog/documentation/pull/16295
---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/agent/guide/how_remote_config_works.md [hunk 1]
```diff
@@ -144,13 +144,12 @@ After you perform these steps, your Agent requests its configuration from Datado
...
 
 
-
-
 {{% /site-region %}}
 
 ## Further Reading
 
 {{< partial name="whats-next/whats-next.html" >}}
+
```
Comment (StefonSimmons) [inline content/en/agent/guide/how_remote_config_works.md:152]: space needed to render link refs as `<a>` tags

**Review by rtrieu — APPROVED** [review]
Thank you for this fix!

**Review by csantosdd — APPROVED** [review]

**Comment by hestonhoffman** [comment]
@StefonSimmons if you fix the merge conflict on this one, I can get it merged in

**Comment by StefonSimmons** [comment]
> @StefonSimmons if you fix the merge conflict on this one, I can get it merged in

@hestonhoffman I believe Kaylyn has a hold on merging any changes to the site-region shortcodes

**Comment by hestonhoffman** [comment]
@StefonSimmons Yes, sorry. I'll check in again when we're ready to merge

File: content/en/logs/error_tracking/browser_and_mobile.md [hunk 2]
```diff
@@ -77,11 +77,11 @@ If you have not setup the Datadog Android SDK yet, follow the [in-app setup inst
...
    ```
 
-[1]: https://app.datadoghq.com/logs/onboarding/client
 [2]:/logs/log_collection/android/#setup
-[3]: https://github.com/Datadog/dd-sdk-android
-[4]: /logs/log_collection/android/?tab=kotlin#setup
 
+[1]: https://app.datadoghq.com/logs/onboarding/client
+[2]: https://github.com/Datadog/dd-sdk-android
+[3]: /logs/log_collection/android/?tab=kotlin#setup
```
Comment (bcaudan) [inline content/en/logs/error_tracking/browser_and_mobile.md:84]: it seems that you have two `[2]` for this section

**Pull request closed without merging**